pyrometheus/__init__.py

```diff
@@ -117,26 +153,70 @@ def _(poly: ct.NasaPoly2, arg_name):
 
     def gen(c, t):
         assert len(c) == 7
-        return (c[0]*log(t)
-                + c[1]*t + c[2]/2*t**2 + c[3]/3*t**3 + c[4]/4*t**4
-                + c[6])
+        return (
+            c[0] * log(t)
+            + c[1] * t
+            + c[2] / 2 * t ** 2
+            + c[3] / 3 * t ** 3
+            + c[4] / 4 * t ** 4
+            + c[6]
+        )
 
     return nasa7_conditional(p.Variable(arg_name), poly, gen)
 
-# }}}
 
+# }}}
 
 # {{{ equilibrium constants
 
-def equilibrium_constants_expr(react: ct.Reaction):
-    return 0*p.Variable("T")
+
+def equilibrium_constants_expr(sol: ct.Solution, react: ct.Reaction):
+
+    reactants = react.reactants
+    products = react.products
+
+    indices_r = [sol.species_index(sp) for sp in reactants]
+    indices_p = [sol.species_index(sp) for sp in products]
+
+    coeffs_r = np.array([reactants[sp] for sp in reactants])
+    coeffs_p = np.array([products[sp] for sp in products])
+
+    n_r = coeffs_r.sum()
+    n_p = coeffs_p.sum()
```

Comment:
Could you give a better variable name for `n_p`? (I know nothing about chemistry, so maybe it's obvious, but I'm curious what units are being dealt with here.)

Reply:
I realized I don't need to compute these in two separate lines, or even there at all. I've made some further, related changes:
1. I've renamed coeffs_r, coeffs_p to nu_r, nu_p. I've added a comment to indicate the nu's are stoichiometric coefficients, which is convention after all.
2. I now have a new variable `sum_nu_net = (nu_p - nu_r).sum()`, which really is all we need.
These will be reflected in the next commit.